server/api/router/post.ts

```diff
@@ -218,13 +218,16 @@ export const postRouter = createTRPCRouter({
                   eq(like.postId, postId),
                   eq(like.userId, ctx.session?.user?.id),
                 ),
-              );
-            await tx
-              .update(post)
-              .set({
-                likes: decrement(post.likes),
-              })
-              .where(eq(post.id, postId));
+              )
+              .returning();
+            if (res.length !== 0) {
```

Comment:
Essentially we are ensuring there is a like to remove before be decrement the like count stored on the post table.

Reply:
I think this should work. Could the same happen in the positive way (eg giving infinite likes?)

Reply:
No because theres a foreign key constraint that prevents a user creating two 'like' rows for the same post. This causes the transaction to fail.

For some reason when you try to delete something that doesnt exist it doesnt throw an error and instead just goes 👍🏻